### DISCUSSION TITLE
What social icons should we add to the list?

### DISCUSSION THREAD

**Discussion by gingervitis** [owncast/owncast#526] [General Owncast discussion]
### Current list of social icons
Currently we have included: (as of 0.0.3)
- bandcamp
- discord
- facebook
- github
- instagram
- linkedin
- mastodon
- patreon
- paypal
- snapchat
- soundcloud
- spotify
- tiktok
- twitch
- twitter
- YouTube

We currently  have requests to add:
- Keyoxide
- Ko-fi
- A general website "link"

.. which we'll add to an upcoming release.

### **So, what platform icons would you like to see in the list?**  

Please let us know, and if you could point us to a high-quality logo of the platform that'd be great! 🤗 

(We'll also work on making the logos higher display at a higher quality as well.)

Thanks!

**Comment by muesli** [comment]
Generic RSS icon would be nice.

**Reply by brown7477** [reply]
MiiPage

**Reply by jeyemwey** [reply]
From the content that I can see at miipage, it looks like it stands against most values that the Owncast Community has committed to as [our standards](https://github.com/owncast/owncast/blob/master/CODE_OF_CONDUCT.md#our-standards). I’m therefore standing against adding MiiPage to our social networks list, just like I wouldn’t want to have Gab or Parler integrated in this project.

**Reply by brown7477** [reply]
I appreciate your feedback but can you explain how MiiPage goes against the Standards set? 

- The use of sexualized language or imagery and unwelcome sexual attention or advances
- Trolling, insulting/derogatory comments, and personal or political attacks
- Public or private harassment
- Publishing others' private information, such as a physical or electronic address, without explicit permission
- Other conduct which could reasonably be considered inappropriate in a professional setting

I am sorry but I am not seeing any of this being done on the site.
Yes, it may be a little geared to the right politically, but it's not disrespectful or hateful.

Anyways, I will accept whatever decision y'all make regarding it.

**Reply by geekgonecrazy** [reply]
At any rate I think the solution proposed below makes total sense as let's anyone add any arbitrary site with any logo. 

This in my opinion is the most flexible and won't require us to add to a code base everyone's logo.

https://github.com/owncast/owncast/discussions/526#discussioncomment-243927

**Comment by gabek** [comment]
Since the Fediverse as a whole has started to adopt us, it might make sense to get the big services.

* Pixelfed
* Peertube
* Funkwhale

**Comment by geekgonecrazy** [comment]
Any way we can adopt a way of discovering the icon of the service?  I don’t remember if such a standard has emerged.  Something like the rel links to identify.. would be nice for discovering icons and social services.  Since they are a dime a dozen now days

**Reply by gabek** [reply]
While it's likely doable by parsing meta tags, I think it would probably cause a lot of unnecessary extra requests to random 3rd parties, slowing things down, and probably failing a lot of the time.

**Reply by SeaLife** [reply]
Especially when the Service is running behind a corporate proxy without internet. :) There should be no need for external services (excluding those, u can self-host ofc)

**Comment by gingervitis** [comment]
gitlab

**Comment by SeaLife** [comment]
Just to get you guys a source for SVG's https://simpleicons.org/ :) #515 :)

Some minor-brands like "Keyoxide" are missing, but the developer got in touch with us in another issue,
he may provide a good SVG to use :)

Apart from those, we could also allow to set a custom image path. This would allow us to
upload the images our selves and set the URLs for them :)

I would ship owncast with the major brands included, so i dont need a icon-path for Github and stuff, but if i wanna have a custom brand, maybe a intern "brand" to a wiki with a custom icon, i could just add a handle for it with an icon-path instead :)

```yaml
socialHandles:
  - platform: internal-wiki
    icon: /img/socialhandles/cool-wiki.svg
    url: https://my-internal-wiki.example.org
  - platform: github
    url: https://github.com/SeaLife
#    icon: /img/socialhandles/github.svg (default, no need to apply this)
```

**Comment by SeaLife** [comment]
I have made some changes to my owncast code to accomplish the following:

![image](https://user-images.githubusercontent.com/6567620/103170224-f62b3400-4842-11eb-91ab-920ec222e233.png)

![image](https://user-images.githubusercontent.com/6567620/103170231-ffb49c00-4842-11eb-830c-1e78fd2c7165.png)


Oppinions?

**Reply by ghost** [reply]
Where do you make this edit?

**Reply by jeyemwey** [reply]
Hello @pa5tlife, these changes were made in the old config format which is no longer used. You can use the admin interface to place your social media links.

**Comment by kaffeeringe** [comment]
www

**Reply by geekgonecrazy** [reply]
Do you mean a website in general?

**Reply by kaffeeringe** [reply]
Yes. The "homepage" for the stream.

**Reply by SeaLife** [reply]
With my suggested change, you could easily use any icon u can imagine. A "www" icon in form of a globe is already there :)

**Comment by gabek** [comment]
Should we have a generic "Fediverse" icon for all the other ActivityPub-supporting services, or should we support all the different fediverse options specifically?
